### PR TITLE
Adds basic support for multiple byond versions to build scripts.

### PR DESCRIPTION
**Pull request by AnturK**
You can now specify few named versions in `tools/build/dm_versions.json` file and either set them as default (overriding anything but env vars) or just use them with new --dm-version parameter.

Makes juggling versions bit easier during testing. I'll look into adding support in sdmm so debugging is in line later.

### CODE REVIEW

**Comment by Mothblocks** [comment]
@stylemistake i dont understand juke build well enough to be able to review this

File: tools/build/lib/byond.js [hunk 1]
```diff
@@ -58,15 +63,69 @@ const getDmPath = async () => {
...
+    return []
+  return [map_entry.path];
+}
+
+
+/** @type {[{ name, path, default }]} */
+let mapFileContents;
+export const NamedVersionFile = "tools/build/dm_versions.json"
+
+const getMapFileContents = (throw_on_fail) => {
```
Comment (stylemistake) [inline tools/build/lib/byond.js:91]: Confusing name, I thought this has to do with byond maps, but this is essentially just a version list.

**Review by stylemistake — APPROVED** [review]
Looks good otherwise

File: tools/build/lib/byond.js (hunk 1 above)
Comment (AnturK) [reply tools/build/lib/byond.js:91]: Sure, will rename this

**Pull request merged**